hc_core/src/logger.rs

```diff
@@ -0,0 +1,24 @@
+/// trait that defines the logging functionality that hc_core requires
+pub trait Logger {
+    fn log(&mut self, msg: String);
+    fn read(&self) -> String;
+}
+
+#[derive(Clone, Debug)]
+pub struct SimpleLogger {
+    //    log: Vec<String>,
+}
```

Comment:
I think it would be worth looking into https://crates.io/crates/log

provides some convenience macros like

```rust
trace!("doing a thing: {}", something);
warn!("something bad: {}", something);
```

and has a plugable backend that we could swap out if needed. could start with https://crates.io/crates/simple_logger

Reply:
Yah, I looked at that this morning.  It works fine for our global debug logging, and I do think we should use it for that, but it doesn't work for per-app logging.  I created a ticket #21 regarding this distinction.  But it certainly will help us distinguish between container level logging and application level logging.